lib/Doctrine/ORM/Id/UuidGenerator.php

```diff
@@ -26,11 +29,17 @@ public function __construct()
 
     /**
      * {@inheritDoc}
+     *
+     * @throws NotSupported
      */
     public function generate(EntityManager $em, $entity)
     {
         $conn = $em->getConnection();
-        $sql  = 'SELECT ' . $conn->getDatabasePlatform()->getGuidExpression();
+        if (method_exists($conn, 'getGuidExpression')) {
```

Comment:
This lines looks wrong to me, for several reasons. If it is indeed wrong, I think that answers your question about unit tests :stuck_out_tongue: